generate.sh

```diff
@@ -0,0 +1,148 @@
+pathToCSV=
+pathToDestRepo="$scriptroot"
+positional_args=()
+requiredOptionSpecified=false
+while :; do
+    if [ $# -le 0 ]; then
+        break
+    fi
+    lowerI="$(echo "$1" | awk '{print tolower($0)}')"
+    case $lowerI in
+        "-?"|-h|--help)
+            usage
+            exit 0
+            ;;
+        --pkgcsv)
+            if [ ! -f "$2" ]; then
+                echo -e "${RED}ERROR: CSV file not found - $2${NC}"
+                exit 1
+            fi
+            pathToCSV="$(cd "$(dirname "$2")"; pwd)/$(basename "$2")"
+            requiredOptionSpecified=true
+            shift
+            ;;
+        --dest)
+            if [ ! -d "$2" ]; then
+                echo -e "${RED}ERROR: dest not found - $2${NC}"
+                exit 1
+            fi
+            pathToDestRepo="$(cd -P "$2" && pwd)"
+            shift
+            ;;
+        --pkg)
+            packageVersion="$2"
+            requiredOptionSpecified=true
+            shift
+            ;;
+        *)
+            echo -e "${RED}Unrecognized argument '$1'${NC}"
+            usage
+            exit 1
+            ;;
+    esac
+
+    shift
+done
+
+if [ "$requiredOptionSpecified" != "true" ]; then
+    usage
+    exit 1
+fi
+
+if [ "$packageVersion" != "" ]; then
+    mkdir -p "$(dirname "$defaultPathToCSV")"
+    echo "$packageVersion" > "$defaultPathToCSV"
+fi
+
+if [ "$pathToCSV" == "" ]; then
+    pathToCSV="$defaultPathToCSV"
+fi
+
+# Generate restore projects for each target package
+# Packges are restored in their own project to eliminate any conflicts
+# between different versions of the same package.
+INPUT="$pathToCSV"
+OLDIFS=$IFS
+IFS=,
+DEFAULT_TFM=net6.0
+restoreProjectsDir="$scriptroot/artifacts/targetRestoreProjects"
+targetPackageTemplate="$scriptroot/src/referencePackageSourceGenerator/GenerateSource/targetPackage.csproj.template"
+if [ -d "$restoreProjectsDir" ]; then
+  rm -rf "$restoreProjectsDir"
+fi
+mkdir -p "$restoreProjectsDir"
+[ ! -f $INPUT ] && { echo "$INPUT file not found"; exit 99; }
+while read pkgName pkgVersion tfm
+do
+    sed "s/##PackageName##/${pkgName}/g" "$targetPackageTemplate" > "$restoreProjectsDir/$pkgName.$pkgVersion.csproj"
+    sed -i "s/##PackageVersion##/${pkgVersion}/g" "$restoreProjectsDir/$pkgName.$pkgVersion.csproj"
+    if [ "$tfm" != "" ]; then
+        sed -i "s/##Tfm##/${tfm}/g" "$restoreProjectsDir/$pkgName.$pkgVersion.csproj"
+    else
+        sed -i "s/##Tfm##/${DEFAULT_TFM}/g" "$restoreProjectsDir/$pkgName.$pkgVersion.csproj"
+    fi
+done < $INPUT
+IFS=$OLDIFS
+
+# Build the projects to generate source and projects
+"$scriptroot/eng/common/build.sh" --build --restore -bl /p:GeneratePackageSource=true /p:GeneratorVersion=2 $@
+
+# Copy source to destination
+pathToGeneratedSrc="$scriptroot/artifacts/generatedSrc"
+pathToRepoSrc="$pathToDestRepo/src/referencePackages/src"
+
+pushd "$pathToGeneratedSrc"
+cp ./ProjectsToBuildAdditions.txt ./CopyProjects.sh
```

Comment:
What is `ProjectsToBuildAdditions.txt` all used for?  This is the only reference I see.  Would it make sense to change the format to be a simple project file list?  It would be more consumable here.

Reply:
In the pre-arcade version of this generation logic, the list of projects to build is manually maintained, so the instructions direct the user to copy from this file and paste into the projects to build props file.  This isn't needed in 6.0 and main, so I modified to just generate the copy script directly.